GitCommands/RevisionReader.cs

```diff
@@ -385,71 +384,46 @@ long ParseUnixDateTime(in ReadOnlySpan<byte> array)
 
             return true;
 
-            void ParseAssert(string? message)
-            {
-                _noOfParseError++;
-                Debug.Assert(_noOfParseError > 1, message);
-                Trace.WriteLineIf(_noOfParseError < 10, message);
-            }
-        }
-
-        #region Nested type: StringLineReader
-
-        /// <summary>
-        /// Simple type to walk along a string, line by line, without redundant allocations.
-        /// </summary>
-        internal ref struct StringLineReader
-        {
-            private readonly ReadOnlySpan<char> _s;
-            private int _index = 0;
-
-            public StringLineReader(in ReadOnlySpan<char> s)
-            {
-                _s = s;
-            }
-
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            public string? ReadLine(in bool useStringPool)
+            string? GetNextLine(ReadOnlySpan<char> s, in bool useStringPool)
             {
-                if (_index >= _s.Length)
+                if (offset >= s.Length)
                 {
                     return null;
                 }
 
-                int lineLength = _s[_index..].IndexOf('\n');
+                int lineLength = s.Slice(offset).IndexOf('\n');
```

Comment:
Isn't there something like
```suggestion
                int lineLength = s.IndexOf(start: offset, '\n');
```
?

Reply:
no
For Span() this is about the same, Slice is basically copying the pointer and adding the offset to start so such methods are not needed